core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -89,8 +90,12 @@
 
     private transient Run<?, ?> run;
 
-    public ParametersAction(List<ParameterValue> parameters) {
-        this.parameters = parameters;
+    /**
+     * Constructs a new action with a specified list of parameter values.
+     * @param parameters Parameter values
+     */
+    public ParametersAction(@CheckForNull List<ParameterValue> parameters) {
```

Comment:
> I do not think there was ever any intent to support a null method parameter here.

Lack of documentation is something causing it. As a Jenkins contributor, I prefer the robust behavior in the core since we have no idea how many OSS/private source plugins are affected